crates/sema/src/typeck/mod.rs

```diff
@@ -11,13 +11,37 @@ pub(crate) fn check(gcx: Gcx<'_>) {
         gcx.sess,
         gcx.hir.par_contract_ids().for_each(|id| {
             check_duplicate_definitions(gcx, &gcx.symbol_resolver.contract_scopes[id]);
+            check_payable_fallback_without_receive(gcx, id);
         }),
         gcx.hir.par_source_ids().for_each(|id| {
             check_duplicate_definitions(gcx, &gcx.symbol_resolver.source_scopes[id]);
         }),
     );
 }
 
+fn check_payable_fallback_without_receive(gcx: Gcx<'_>, contract_id: hir::ContractId) {
+    let mut payable_fallback_found = false;
+    let mut receive_found = false;
+    for item in gcx.hir.contract_items(contract_id) {
```

Comment:
fallback and receive are stored in contract directly, don't need to scan for them

Reply:
✅